File: python/algos/remove_element.py
from typing import List


def remove_element(nums: List[int], value: int) -> int:
    """Removes value from nums in place."""
    i = 0
    for x in nums:
        if x != value:
            nums[i] = x
            i += 1
    return i


if __name__ == "__main__":
    assert remove_element([3, 2, 2, 3], 3) == 2
    assert remove_element([0, 1, 2, 2, 3, 0, 4, 2], 2) == 5

File: rust/src/algos/remove_element.rs
pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
    let mut i = 0;
    for index in 0..nums.len() {
        if nums[index] != val {
            nums[i] = nums[index];
            i += 1;
        }
    }
    i as i32
}
